src/Authority/Authority.php

```diff
@@ -108,26 +108,38 @@ public function cannot($action, $resource, $resourceValue = null)
     /**
      * Define privilege for a given action and resource
      *
-     * @param string        $action Action for the rule
+     * @param array|string  $action Action for the rule
      * @param mixed         $resource Resource for the rule
      * @param Closure|null  $condition Optional condition for the rule
      * @return Rule
      */
     public function allow($action, $resource, $condition = null)
     {
+        if (is_array($action)) {
```

Comment:
The way it's currently implemented it allows for infinite nesting of arrays of rules since it recurses.  Was this intentional?   Otherwise I'd recommend nixing the type check and simply cast the var to an array and iterate over it, then call `$this->addRule` internally in the loop.

I'm also interested in what you think the appropriate return type should be if multiple rules are processed.  Do you think it'd be beneficial to return a single rule if a single action is created and a collection if multiple are passed in, or always return a collection regardless of the number of actions passed in?